flow-decomposition/src/test/java/com/powsybl/flow_decomposition/FlowDecompositionTests.java

```diff
@@ -196,6 +196,15 @@ void testFlowDecompositionOnHvdcNetwork() {
         assertTrue(flowDecompositionResults.getZoneSet().contains(Country.NL));
     }
 
+    @Test
+    void testConnectedComponentModeChangesFromAllToMain() {
+        Network network = TestUtils.getMicroGridNetworkWithBusBarSectionOnly();
+        LoadFlowParameters loadFlowParameters = new LoadFlowParameters().setConnectedComponentMode(LoadFlowParameters.ConnectedComponentMode.ALL);
+        FlowDecompositionComputer flowDecompositionComputer = new FlowDecompositionComputer(new FlowDecompositionParameters(), loadFlowParameters);
+        flowDecompositionComputer.run(new XnecProviderAllBranches(), network);
+        assertEquals(LoadFlowParameters.ConnectedComponentMode.MAIN, loadFlowParameters.getConnectedComponentMode());
```

Comment:
Please to not rely on side effect for testing this.
I would rather like that you expose a protected method and test this method with a unit test.

Reply:
I didn't think of it, thanks!